src/lib/content/tab-script.js

```diff
@@ -309,10 +328,12 @@ export default class TabScript extends UIStateAPI {
   }
 
   static readObject (jsonObject) {
-    let tabScript = new TabScript(jsonObject.tabID)
+    let tabScript = new TabScript(jsonObject.tabObj)
 
     for (let prop of TabScript.symbolProps) {
-      if (jsonObject.hasOwnProperty(prop)) { tabScript[prop] = Symbol.for(jsonObject[prop]) }
+      if (jsonObject.hasOwnProperty(prop)) {
+        tabScript[prop] = (typeof jsonObject[prop] === 'symbol') ? jsonObject[prop] : Symbol.for(jsonObject[prop])
```

Comment:
Yes, we need this check because this method readObject use json (from message.body) and there are no symbols there ( I didn't remove it completely because may be there are some situations where it could be passed as Symbols).

Symbol properties are passed this way
![image](https://user-images.githubusercontent.com/12377640/43456008-2e0223d2-9505-11e8-8b07-ae937b73ce34.png)

Reply:
If there are no symbol props in a message body (i.e. no props with names listed in `symbolProps` array), then `for (let prop of TabScript.symbolProps) { }` will never be executed as prop names in a body would not be within a `symbolProps` array.

If there are any symbol props (listed in `symbolProps` array), then those be symbols for sure and we can simply do
```
 for (let prop of TabScript.symbolProps) {
      if (jsonObject.hasOwnProperty(prop)) { tabScript[prop] = Symbol.for(jsonObject[prop]) }
}
```
Would the code above pass your tests as it is or do you see any problems with it?